### PR TITLE
Round of deprecation removal

### PR DESCRIPTION
**Pull request by 1ucian0**
This PR removes some methods and functions that they were tagged to be deprecated post-0.9:

- A lot of tuple-bit support, including in layout and in parameter expansion for gates.
- Basis set as plain string.
- `as_dict` in the base validator

### CODE REVIEW

**Comment by mtreinish** [comment]
If we're really planning to push a 0.10 release within the next couple of weeks I'd say we probably should hold off on this. It's been basically 1 month since the 0.9 release which deprecated a bunch of this stuff. I don't think is enough of a transition window for people.

**Comment by mtreinish** [comment]
See: https://github.com/Qiskit/qiskit-terra/pull/3191

**Comment by 1ucian0** [comment]
I moved it to 0.11

**Comment by mtreinish** [comment]
Just marking this as on hold now that the official deprecation policy has merged. This will be eligible for merging on Nov 22nd

**Comment by 1ucian0** [comment]
> But I just noticed that there are no release notes here

Fixed in d76cce0f081d3167fe84c2c6d0b234abff785fe2

File: releasenotes/notes/deprecation_0.9-0163a90875b53832.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+upgrade:
+  - |
```
Comment (mtreinish) [inline releasenotes/notes/deprecation_0.9-0163a90875b53832.yaml]: What upgrade entries for: `Basis set as plain string.` and `as_dict in the base validator`?

Comment (1ucian0) [reply releasenotes/notes/deprecation_0.9-0163a90875b53832.yaml]: Done in 0787dd910

**Comment by 1ucian0** [comment]
Removing `on hold`, since it's time to remove.

**Pull request merged**